test/verify/machineslib.py

```diff
@@ -178,13 +178,41 @@ def prepareStorageDeviceOnISCSI(m, target_iqn, orig_iqn=None):
 class TestMachines(NetworkCase):
     created_pool = False
     provider = None
+    tmp_storage = "/mnt/libvirt"
 
     def setUp(self):
         # HACK: fix this to get along with NetworkCase.setUp(), and use super()
         MachineCase.setUp(self)
+        m = self.machine
+
+        # Prepare tmp directory
+        m.execute("mkdir {0}".format(self.tmp_storage))
+        self.addCleanup(m.execute, "rm -rf {0}".format(self.tmp_storage))
+
+        # Keep pristine state of libvirt
+        orig = "/var/lib/libvirt"
+        backup = os.path.join(self.tmp_storage, "libvirt.backup")
+        m.execute("cp -a {0} {1}".format(orig, backup))
+        self.addCleanup(m.execute, "rm -rf {0} && cp -a {1} {0}".format(orig, backup))
```

Comment:
This means tests work over the original data in /var, and you have to copy over stuff twice. If you bind-mount the backup over the original, then you'll never touch the original bits in /var, tests mess up the backup in tmp_storage instead, and the cleanup cp -a can be avoided (and be replaced with an umount).

Anyway, good enough if this works and doesn't take too much time.